crates/cheatnet/src/rpc.rs

```diff
@@ -28,11 +28,30 @@ use starknet_api::{
     transaction::Calldata,
 };
 
+#[derive(Debug, Default)]
+pub struct UsedResources {
+    pub execution_resources: ExecutionResources,
+    pub l2_to_l1_payloads_length: Vec<usize>,
+}
+
+impl UsedResources {
+    pub fn extend(self: &mut UsedResources, other: &UsedResources) -> &Self {
```

Comment:
Does it have to return &Self?

Reply:
We want to return the same object

Reply:
I don;t think so
```
write_call_contract_response
```

has mutable reference to it and extends the struct

Reply:
In the forge_runtime_extension/mod.rs